BootloaderCorePkg/Stage2/Stage2Support.c

```diff
@@ -667,6 +677,34 @@ BuildExtraInfoHob (
     PlatformUpdateHobInfo (&gLoaderPlatformInfoGuid, LoaderPlatformInfo);
   }
 
+  if ((FixedPcdGetBool (PcdPreOsCheckerEnabled)) && (PldPtr[10] == EFI_FVH_SIGNATURE)) {
```

Comment:
Expecting pre-OS checker to be part of the payload/OsLoader when it is built as an FV image